static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -53,88 +46,122 @@ define([
                         : getClientAdHeight();
                 };
 
+                var topAdRenderedPromise = new Promise(function (resolve) {
+                    mediator.on('modules:commercial:dfp:rendered', function (event) {
+                        if (event.slot.getSlotElementId() === adId) {
+                            resolve();
+                        }
+                    });
+                });
+
+                var newRubiconAdHeightPromise = new Promise(function (resolve) {
+                    window.addEventListener('message', function (event) {
+                        var data = JSON.parse(event.data);
+                        var $iframe = getAdIframe();
+                        var isRubiconAdEvent = data.type === 'set-ad-height';
```

Comment:
Do we ever have multiple rubicon ads running on a page? We might need to do something to differentiate the messages' source frames if we do. (I hope not, because this is really awkward when you don't have control over the iframe)